File: backend/src/RepoFluent.Application/VersionComparison.cs
namespace RepoFluent.Application;

public sealed record VersionComparison(
    Guid BaseVersionId,
    Guid TargetVersionId,
    string PackageId,
    string BaseVersion,
    string TargetVersion,
    IReadOnlyList<VersionChange> Changes,
    IReadOnlyList<string> AffectedObjectiveIds,
    bool LearnerRefreshRequired);

File: frontend/projects/api/src/lib/version-comparison.ts
import type { VersionChange } from './version-change';

export interface VersionComparison {
  baseVersionId: string;
  targetVersionId: string;
  packageId: string;
  baseVersion: string;
  targetVersion: string;
  changes: VersionChange[];
  affectedObjectiveIds: string[];
  learnerRefreshRequired: boolean;
}
